docs/Questions.md

```diff
@@ -0,0 +1,59 @@
+<H1> Folder structure(Pages, Components, Templates, Organisms, Molecules, Atoms) </H1>
+We should stick to one way of organizing our folder and component's structure.
+Utils have files with .tsx file extension, which I think is inappropriate.
+We have validation.js and wdyr.js files at the top level of src
+
+<H1> Styles </H1>
+Currently we both use Styled-Components and scss across our codebase. I think we should stick to one and remove the other
+
+<H1> Data fetching & selectors in components </H1>
+We have a lot of ambiguity about how we interact with data in our app.
+For example:
+
+- `useConnects` in components with different ways of solving the issue, when there is no needed data(like a venue id)
+- firestore direct pushes in `actions` and components
```

Comment:
Agreed. I would like to see us move all `firestore` calls out of component files.

For the 'low level' firestore queries/etc, I would like to see them in an `api/` folder or similar, and expose a higher level 'interface' that other places in the app use (eg. instead of `firestore.collection('users').foo.bar.baz`, we would just call `fetchUsers` or something similar) This (along with the selector functions, and wrapping data fetching/usage in custom hooks) will allow us to centralise our data store patterns/decouple the rest of the codebase from them, which will make it far easier to change them in future (eg. if we choose to migrate away from firestore, etc)

Updates/similar should probably be centralised with Redux, and then components would only dispatch redux actions.

I like the pattern of using Redux Saga (or potentially Redux Observable) to 'watch' for events, and then 'react' by requesting/updating data/etc.

- https://redux-saga.js.org/ (this is what I have used in previous projects, so am more familiar with)
- https://redux-observable.js.org/

I tend to have a fairly strong preference of not using `redux-thunk` directly, as it doesn't allow nice/easy debugging/etc with things like redux devtools.